fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/digitalcash/TransactionObjectBuilder.java

```diff
@@ -0,0 +1,72 @@
+package com.github.dedis.popstellar.model.objects.digitalcash;
+
+import com.github.dedis.popstellar.model.objects.Channel;
+import com.github.dedis.popstellar.model.objects.InputObject;
+import com.github.dedis.popstellar.model.objects.OutputObject;
+
+import java.util.List;
+
+public class TransactionObjectBuilder {
+
+  private Channel channel;
+
+  // version
+  private int version;
+
+  // inputs
+  private List<InputObject> inputs;
+
+  // outputs
+  private List<OutputObject> outputs;
+
+  // lock_time
+  private long lockTime;
+
+  private String transactionId;
+
+  public TransactionObjectBuilder setChannel(Channel channel) {
+    this.channel = channel;
+    return this;
+  }
+
+  public TransactionObjectBuilder setVersion(int version) {
+    this.version = version;
+    return this;
+  }
+
+  public TransactionObjectBuilder setInputs(List<InputObject> inputs) {
+    this.inputs = inputs;
+    return this;
+  }
+
+  public TransactionObjectBuilder setOutputs(List<OutputObject> outputs) {
+    this.outputs = outputs;
+    return this;
+  }
+
+  public TransactionObjectBuilder setLockTime(long lockTime) {
+    this.lockTime = lockTime;
+    return this;
+  }
+
+  public TransactionObjectBuilder setTransactionId(String transactionId) {
+    this.transactionId = transactionId;
+    return this;
+  }
+
+  public TransactionObject build() {
+    if (channel == null) {
+      throw new IllegalStateException("channel is null");
+    }
+    if (inputs == null) {
+      throw new IllegalStateException("inputs is null");
+    }
+    if (outputs == null) {
+      throw new IllegalStateException("outputs is null");
+    }
+    if (transactionId == null) {
+      throw new IllegalStateException("transactionId is null");
+    }
+    return new TransactionObject(channel, version, inputs, outputs, lockTime, transactionId);
```

Comment:
Should `inputs` and `outputs` also be checked if their `null`? It feels a bit weird that only some arguments that are provided to the `TransactionObject` are checked.

Reply:
Should we check if empty list for input and output ?

Reply:
@Tesa27 is it not done on lines 64 and 61 ?

Reply:
@zoemarin, I do not know. Is it completely impossible to have an empty input or output ?

Reply:
@Warhedgehog Sorry I am blind, all good here